aws/resource_aws_docdb_cluster_parameter_group.go

```
@@ -0,0 +1,282 @@
+
+	return resourceAwsDocDBClusterParameterGroupUpdate(d, meta)
+}
+
+func resourceAwsDocDBClusterParameterGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).docdbconn
+
+	describeOpts := &docdb.DescribeDBClusterParameterGroupsInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+	}
+
+	describeResp, err := conn.DescribeDBClusterParameterGroups(describeOpts)
+	if err != nil {
+		if isAWSErr(err, docdb.ErrCodeDBParameterGroupNotFoundFault, "") {
+			log.Printf("[WARN] DocDB Cluster Parameter Group (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+	}
+
+	if len(describeResp.DBClusterParameterGroups) != 1 ||
+		*describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupName != d.Id() {
+		return fmt.Errorf("Unable to find Cluster Parameter Group: %#v", describeResp.DBClusterParameterGroups)
+	}
+
+	arn := aws.StringValue(describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupArn)
+	d.Set("arn", arn)
+	d.Set("description", describeResp.DBClusterParameterGroups[0].Description)
+	d.Set("family", describeResp.DBClusterParameterGroups[0].DBParameterGroupFamily)
+	d.Set("name", describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupName)
+
+	describeParametersOpts := &docdb.DescribeDBClusterParametersInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+		Source:                      aws.String("user"),
+	}
+
+	describeParametersResp, err := conn.DescribeDBClusterParameters(describeParametersOpts)
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("parameter", flattenDocDBParameters(describeParametersResp.Parameters)); err != nil {
+		return fmt.Errorf("error setting docdb cluster parameter: %s", err)
+	}
+
+	resp, err := conn.ListTagsForResource(&docdb.ListTagsForResourceInput{
+		ResourceName: aws.String(arn),
+	})
+
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
```

We should return an error here instead of a debug log:

```suggestion
		return fmt.Errorf("error listing tags for DocDB Cluster Parameter Group (%s): %s", d.Id(), err)
```

---

Fixed